packages/web-components/src/components/ic-checkbox/ic-checkbox.tsx

```diff
@@ -178,6 +184,26 @@ export class Checkbox {
     this.checkboxEl?.focus();
   }
 
+  private getCheckboxGroupId() {
+    return `ic-checkbox-${this.label}-${this.groupLabel}`.replace(/ /g, "-");
+  }
+
+  private hasLabelSlot() {
+    this.labelSlot = this.el.querySelector('[slot="label"]');
+    return !!this.labelSlot;
+  }
+
+  private arrangeLabel() {
+    if (this.labelSlot) {
+      this.label = this.labelSlot.textContent?.trim() || "";
+      setStyles(this.labelSlot, {
+        color: "var(--ic-checkbox-text)",
```

Comment:
I think we might need to check here whether the checkbox is disabled, and if it is, set it to the disabled text colour